spec/index.bs

```diff
@@ -2517,6 +2561,112 @@ dictionary GPUImageBitmapCopyView {
+    ```
+
+  The following validation rules apply:
+
+  For |encoder|:
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPURenderPassEncoder}}
+    is active on |encoder|.
+  - |encoder|.{{GPUCommandEncoder/copyBufferToTexture()}} must not be called when a {{GPUComputePassEncoder}}
+    is active on |encoder|.
+
+  For |source|:
+  - |source| must be [=valid=].
+  - |source|.{{GPUBufferCopyView/buffer}}.{{GPUBuffer/[[usage]]}} must contain {{GPUBufferUsage/COPY_SRC}}.
+
+  For |destination|:
+  - |destination| must be [=valid=].
+  - |destination|.{{GPUTextureCopyView/texture}}.{{GPUTexture/[[textureUsage]]}} must contain
+    {{GPUTextureUsage/COPY_DST}}.
+
+  For |copySize|:
+  - |copySize|.[=Extent3D/width=] cannot be 0.
+  - |copySize|.[=Extent3D/height=] cannot be 0.
+  - |copySize|.[=Extent3D/depth=] cannot be 0.
+  - If the {{GPUTexture/[[dimension]]}} of |destination|.{{GPUTextureCopyView/texture}} is
+    {{GPUTextureDimension/1d}}:
+    - Both |copySize|.[=Extent3D/height=] and [=Extent3D/depth=] must be 1.
+  - If the {{GPUTexture/[[dimension]]}} of |destination|.{{GPUTextureCopyView/texture}} is
+    {{GPUTextureDimension/2d}}:
+    - |copySize|.[=Extent3D/depth=] must be 1.
+
+  For the copy being in-bounds:
+  - If |source|.{{GPUBufferCopyView/rowsPerImage}} is not 0, it must be greater than or equal to
+    |copySize|.[=Extent3D/height=].
+  - (|source|.{{GPUBufferCopyView/offset}} &add; |required bytes in copy|) must not overflow a {{GPUSize64}}.
+  - (|source|.{{GPUBufferCopyView/offset}} &add; |required bytes in copy|) must be smaller than or equal to
+    |source|.{{GPUBufferCopyView/buffer}}.{{GPUBuffer/[[size]]}}.
+  - If the {{GPUTexture/[[dimension]]}} of |destination|.{{GPUTextureCopyView/texture}} is
+    {{GPUTextureDimension/2d}}:
+    - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/x}} &add; |copySize|.[=Extent3D/width=]) must be less than or equal to the [=Extent3D/width=] of the [=physical size=] of |destination|{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}}.
+    - (|destination|.{{GPUTextureCopyView/origin}}.{{GPUOrigin3DDict/y}} &add; |copySize|.[=Extent3D/height=]) must be less than or equal to the [=Extent3D/height=] of the [=physical size=] of |destination|.{{GPUTextureCopyView/texture}} [=subresource=] at [=mipmap level=] {{GPUTextureCopyView/mipLevel}}
+
+  For the texel block alignments:
+  - |source|.{{GPUBufferCopyView/rowsPerImage}} must be a multiple of |blockHeight|.
+  - |copySize|.[=Extent3D/width=] must be a multiple of |blockWidth|.
+  - |copySize|.[=Extent3D/height=] must be a multiple of |blockHeight|.
+  - |source|.{{GPUBufferCopyView/offset}} must be a multiple of |blockSize|.
+
+  For other members in |source|:
+  - If |source|.{{GPUBufferCopyView/bytesPerRow}} is not 0, it must be greater than or equal to the number
+    of |bytes in a complete row|.
```

Comment:
If bytesPerRow=0 is no longer special, this becomes "If height > 1, bytesPerRow must be..."